ui/library/src/components/thumbnails/ThumbnailList.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+
+import { DocumentContext } from '../../context/DocumentContext';
+import { Thumbnail } from './Thumbnail';
+
+type Props = any;
+
+export const ThumbnailList: React.FunctionComponent<Props> = () => {
+  const { numPages } = React.useContext(DocumentContext);
+
+  return (
+    <div className="reader__thumbnail-list">
+      <ul className="reader__thumbnail-list__list">
+        {Array.from({ length: numPages })
+          .map((_, i) => i + 1)
+          .map(pageNumber => (
+            <li key={pageNumber} className="reader__thumbnail-list__item">
+              <Thumbnail pageNumber={pageNumber} />
+            </li>
+          ))}
```

Comment:
Running an array two times looks odd to me, especially your intention is simply converting index to page numbers
```suggestion
          .map(pageIndex => (
            <li key={pageIndex} className="reader__thumbnail-list__item">
              <Thumbnail pageNumber={pageIndex + 1} />
            </li>
          ))}
```

Reply:
@yensung from what i understand if we directly use pageIndex it will not understand the type of pageIndex or considered it to be null.

Reply:
@yensung also the first map from what im understanding is piping through those value to be a number and second one being used to pass it with a type

Reply:
@yensung updated to be one page map.